### PR TITLE
feat: modernize launch splash

### PR DESCRIPTION
**Pull request by PastaPastaPasta**
## Issue being fixed or feature implemented
- Modernize the splash screen visual design and add a progress bar so users have feedback on initialization progress instead of only a static status message.

### Old
Screenshot 2026-03-11 at 23 24 23

### New

https://github.com/user-attachments/assets/aebe03dd-5484-445b-b4e4-0ffb7e57c26e

## What was done?

**Commit 1: Visual modernization**
- Replaced the flat rectangular splash with a modern card-style design featuring rounded corners and a translucent background
- Redesigned layout: smaller centered logo, refined typography (28pt title, 12pt version), subtle text opacity
- Replaced the rectangular network badge with a pill-shaped rounded badge
- Added canvas padding around the card for compositor-based visual separation
- Scoped `QPainter` lifetime to release the pixmap before timer/signal setup

**Commit 2: Phase-based progress bar**
- Added a thin progress bar at the bottom of the splash card that tracks init phases
- Built a phase weight table mapping init messages to progress ranges (e.g., "Loading block index…" = 3–75%)
- Phases with `ShowProgress` callbacks (block verification, replaying, rescanning) interpolate linearly within their range
- Phases without sub-progress use exponential time-based fill (slow curve for long phases like rescan)
- "Verifying wallet(s)…" and "Loading wallet…" are batched as a single bar (verify = 0–20%, load = 20–100%)
- "Rescanning…" resets the bar and uses a very slow exponential curve (tau=120s) appropriate for operations that can take hours
- Snaps to 100% on "Done loading" since the splash is destroyed immediately after
- Phase lookup translates keys via `_()` at runtime so matching works in all locales
- `messageColor` is `const` and captured by value in signal lambdas for thread-safe cross-thread callbacks
- Animation timer deferred until first message arrives to avoid idle repaints

## How Has This Been Tested?
- Manual testing on macOS: verified splash screen appearance, progress bar animation through full startup sequence including wallet verification and loading
- Verified progress bar behavior with `--rescan` flag (bar resets and uses slow curve)
- Confirmed "Done loading" snaps bar to 100% before splash dismissal
- Verified no regressions in testnet/devnet badge rendering

## Breaking Changes
None. Visual-only changes to the splash screen with no impact on consensus, RPC, or wallet behavior.

## Checklist:
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation
- [ ] I have assigned this pull request to a milestone _(for repository code-owners and collaborators only)_

### CODE REVIEW

File: src/qt/splashscreen.cpp [hunk 1]
```diff
@@ -22,16 +22,65 @@
...
+struct PhaseInfo {
+    const char* msg_key;    // untranslated init message (translated at lookup time)
+    qreal start;            // overall progress at phase start (0.0-1.0)
+    qreal end;              // overall progress at phase end (0.0-1.0)
+    bool resetsBar;         // if true, resets the bar to show this phase's own 0-100%
+    bool snapsToEnd;        // if true, instantly completes the bar (for "Done loading")
+};
+
+static const PhaseInfo PHASE_TABLE[] = {
+    {"Loading P2P addresses…",    0.00, 0.02, false, false},
```
Comment (knst) [inline src/qt/splashscreen.cpp:55]: should be `tr()` string

Comment (PastaPastaPasta) [reply src/qt/splashscreen.cpp:55]: Claude says:
"""
These strings are intentionally stored as untranslated const char* keys and translated at lookup time via _(phase.msg_key).translated (line 73). This is the Bitcoin Core i18n pattern — _()
  wraps bilingual_str which goes through the gettext translation system. Using Qt's tr() would be wrong here because:
  1. The strings originate from the core (non-Qt) layer via bilingual_str/_()
  2. They need to match messages that come from non-Qt code (init.cpp, wallet.cpp)
  3. The current approach correctly translates at lookup time

  The design is documented in the comment on line 40-41: "Keys are untranslated strings passed through _() at lookup time so that phase matching works correctly regardless of the active locale."
"""

**Comment by UdjinM6** [comment]
pls see https://github.com/UdjinM6/dash/commits/pr-7210-attempt1/

**Review by thepastaclaw — COMMENTED** [review]
## Code Review

Well-crafted visual modernization with a thoughtful phase-based progress design. No blocking issues. The primary concern is the exponential time constant (tau=5s) causing the bar to appear stuck during long-running phases like block index loading, with a secondary concern about multi-wallet loading visibility.

*Reviewed commit: da5e9c12*

🟡 2 suggestion(s) | 💬 1 nitpick(s)

<details>
<summary>🤖 Prompt for all review comments with AI agents</summary>

```
These findings are from an automated code review. Verify each finding against the current code and only fix it if needed.

In `src/qt/splashscreen.cpp`:
- [SUGGESTION] lines 282-283: Exponential tau=5s fills the bar too quickly for long-running phases
  The time-based fill `1 - exp(-elapsed/5)` reaches 95% of a phase's range in ~15 seconds. For 'Loading block index…' (range 0.03–0.75, the largest time-based phase), this means the bar hits ~0.68 at 15s and asymptotically caps at ~0.71, appearing stuck for the remainder of what can be a multi-minute operation on nodes with large blockchains (500k+ blocks). A tau of 30–60s would spread the visual fill more evenly across realistic load times, or the tau could be scaled per-phase based on the range width.
- [SUGGESTION] line 64: Multi-wallet 'Loading wallet…' progress is invisible for the second wallet onward
  'Loading wallet…' has `resetsBar=false`, and `initMessage()` sends the identical string `_("Loading wallet…").translated` for every wallet (src/wallet/load.cpp:129). Since 'Verifying wallet(s)…' fires once (load.cpp:56) — not per-wallet — consecutive 'Loading wallet…' messages have the same phase pointer AND the same message text, so `phase_changed` is false. The `phaseTimer` is never restarted, the time-based fill is already near its cap from the first wallet, and `displayProgress` stays put. The second wallet's loading time is invisible to the user.

Consider either setting `resetsBar=true` for this phase (matching how 'Rescanning…' handles multi-wallet), or prefixing the wallet name to the initMessage so the `message != m_current_phase_message` check triggers a phase restart.
```
</details>

File: src/qt/splashscreen.cpp [hunk 2]
```diff
@@ -166,38 +269,61 @@ bool SplashScreen::eventFilter(QObject * obj, QEvent * ev) {
...
+    if (curProgress >= 0) {
+        // When a phase reports real sub-progress, map that 0-100 value into the
+        // phase's assigned range directly.
+        return phaseStart + (phaseEnd - phaseStart) * (curProgress / 100.0);
+    }
+    // Otherwise fall back to a time-based curve so phases driven only by
+    // initMessage() still show movement without claiming exact progress.
+    qreal elapsed = phaseTimer.elapsed() / 1000.0;
+    qreal fraction = 1.0 - std::exp(-elapsed / 5.0);
+    return phaseStart + (phaseEnd - phaseStart) * fraction * EXPONENTIAL_FILL_CAP;
```
Comment (thepastaclaw) [inline src/qt/splashscreen.cpp:283]: **🟡 Suggestion: Exponential tau=5s fills the bar too quickly for long-running phases**

The time-based fill `1 - exp(-elapsed/5)` reaches 95% of a phase's range in ~15 seconds. For 'Loading block index…' (range 0.03–0.75, the largest time-based phase), this means the bar hits ~0.68 at 15s and asymptotically caps at ~0.71, appearing stuck for the remainder of what can be a multi-minute operation on nodes with large blockchains (500k+ blocks). A tau of 30–60s would spread the visual fill more evenly across realistic load times, or the tau could be scaled per-phase based on the range width.

<sub>source: ['claude']</sub>

<details>
<summary>🤖 Fix this with AI agents</summary>

```
These findings are from an automated code review. Verify each finding against the current code and only fix it if needed.

In `src/qt/splashscreen.cpp`:
- [SUGGESTION] lines 282-283: Exponential tau=5s fills the bar too quickly for long-running phases
  The time-based fill `1 - exp(-elapsed/5)` reaches 95% of a phase's range in ~15 seconds. For 'Loading block index…' (range 0.03–0.75, the largest time-based phase), this means the bar hits ~0.68 at 15s and asymptotically caps at ~0.71, appearing stuck for the remainder of what can be a multi-minute operation on nodes with large blockchains (500k+ blocks). A tau of 30–60s would spread the visual fill more evenly across realistic load times, or the tau could be scaled per-phase based on the range width.
```
</details>

File: src/qt/splashscreen.cpp [hunk 3]
```diff
@@ -22,16 +22,72 @@
...
+    {"Loading P2P addresses…",    0.00, 0.02, false, false},
+    {"Loading banlist…",          0.02, 0.03, false, false},
+    {"Loading block index…",      0.03, 0.75, false, false},
+    {"Verifying blocks…",         0.75, 0.88, false, false},
+    {"Replaying blocks…",         0.75, 0.88, false, false},
+    {"Pruning blockstore…",       0.88, 0.90, false, false},
+    {"Starting network threads…", 0.90, 0.94, false, false},
+    {"Rescanning…",               0.00, 1.00, true,  false},
+    {"Verifying wallet(s)…",      0.00, 0.20, true,  false},
+    {"Loading wallet…",           0.20, 1.00, false, false},
```
Comment (thepastaclaw) [inline src/qt/splashscreen.cpp:64]: **🟡 Suggestion: Multi-wallet 'Loading wallet…' progress is invisible for the second wallet onward**

'Loading wallet…' has `resetsBar=false`, and `initMessage()` sends the identical string `_("Loading wallet…").translated` for every wallet (src/wallet/load.cpp:129). Since 'Verifying wallet(s)…' fires once (load.cpp:56) — not per-wallet — consecutive 'Loading wallet…' messages have the same phase pointer AND the same message text, so `phase_changed` is false. The `phaseTimer` is never restarted, the time-based fill is already near its cap from the first wallet, and `displayProgress` stays put. The second wallet's loading time is invisible to the user.

Consider either setting `resetsBar=true` for this phase (matching how 'Rescanning…' handles multi-wallet), or prefixing the wallet name to the initMessage so the `message != m_current_phase_message` check triggers a phase restart.

<sub>source: ['claude']</sub>

<details>
<summary>🤖 Fix this with AI agents</summary>

```
These findings are from an automated code review. Verify each finding against the current code and only fix it if needed.

In `src/qt/splashscreen.cpp`:
- [SUGGESTION] line 64: Multi-wallet 'Loading wallet…' progress is invisible for the second wallet onward
  'Loading wallet…' has `resetsBar=false`, and `initMessage()` sends the identical string `_("Loading wallet…").translated` for every wallet (src/wallet/load.cpp:129). Since 'Verifying wallet(s)…' fires once (load.cpp:56) — not per-wallet — consecutive 'Loading wallet…' messages have the same phase pointer AND the same message text, so `phase_changed` is false. The `phaseTimer` is never restarted, the time-based fill is already near its cap from the first wallet, and `displayProgress` stays put. The second wallet's loading time is invisible to the user.

Consider either setting `resetsBar=true` for this phase (matching how 'Rescanning…' handles multi-wallet), or prefixing the wallet name to the initMessage so the `message != m_current_phase_message` check triggers a phase restart.
```
</details>

File: src/qt/splashscreen.cpp [hunk 4]
```diff
@@ -22,16 +22,72 @@
...
+    {"Done loading",              0.94, 1.00, false, true},
+};
```
Comment (thepastaclaw) [inline src/qt/splashscreen.cpp:66]: **💬 Nitpick: Phase-based progress state machine has no unit tests**

The phase lookup, progress calculation, phase transition logic, and multi-wallet edge cases form a non-trivial state machine. Extracting the progress logic (phase table, calcOverallProgress, phase transition in showMessage) into a testable non-Qt class would allow testing the state transitions and multi-wallet scenarios without a full Qt test harness. Low priority given this is non-consensus GUI code, but would help catch regressions as the phase table evolves.

<sub>source: ['claude']</sub>

**Review by UdjinM6 — APPROVED** [review]
light ACK da5e9c12129c4ff2a91cc0b51c7efc1f4268c74f

**Review by knst — APPROVED** [review]
LGTM 3d45e615ca20be133472c7a1d6deef5fa3bca3ed
